app/src/ereferrals/java/org/eyeseetea/malariacare/strategies/DashboardActivityStrategy.java

```diff
@@ -15,20 +23,45 @@
 import org.eyeseetea.malariacare.data.database.utils.PreferencesEReferral;
 import org.eyeseetea.malariacare.data.database.utils.PreferencesState;
 import org.eyeseetea.malariacare.data.database.utils.Session;
+import org.eyeseetea.malariacare.data.io.FileDownloader;
+import org.eyeseetea.malariacare.data.net.ConnectivityManager;
+import org.eyeseetea.malariacare.data.repositories.MediaRepository;
+import org.eyeseetea.malariacare.domain.boundary.IConnectivityManager;
+import org.eyeseetea.malariacare.domain.boundary.executors.IAsyncExecutor;
+import org.eyeseetea.malariacare.domain.boundary.io.IFileDownloader;
 import org.eyeseetea.malariacare.domain.boundary.repositories.ICredentialsRepository;
 import org.eyeseetea.malariacare.domain.entity.UIDGenerator;
+import org.eyeseetea.malariacare.domain.exception.LoadingNavigationControllerException;
 import org.eyeseetea.malariacare.domain.usecase.GetUrlForWebViewsUseCase;
+import org.eyeseetea.malariacare.domain.usecase.pull.DownloadMediaUseCase;
 import org.eyeseetea.malariacare.fragments.DashboardUnsentFragment;
 import org.eyeseetea.malariacare.fragments.WebViewFragment;
-import org.eyeseetea.malariacare.domain.exception.LoadingNavigationControllerException;
 import org.eyeseetea.malariacare.layout.adapters.survey.navigation.NavigationBuilder;
+import org.eyeseetea.malariacare.presentation.executors.AsyncExecutor;
 
+import java.util.HashMap;
 
 public class DashboardActivityStrategy extends ADashboardActivityStrategy {
+    public static final int REQUEST_GOOGLE_PLAY_SERVICES = 102;
+
+    static final int REQUEST_AUTHORIZATION = 101;
+
+    public interface Callback {
```

Comment:
DashboardActivityStrategy should not define the callback, this should be define inside of use case and has no external dependencies for example:

public interface Callback{
  void onError(FileDownloadException ex);
  void onSuccess(HashMap<String, String> syncedFiles);
}

we can create FileDownloadException for group problems and use cause property to send encapsulated exceptions as UserRecoverableAuthIOException, GooglePlayServicesAvailabilityIOException ...

public class FileDownloadException extends Exception{
     public FileDownloadException(String message, Throwable cause) {
        super(message, cause);
    }
}